hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -68,12 +71,30 @@
 
   private static Logger logger = LogManager.getLogger(HBaseIndex.class);
   private static Connection hbaseConnection = null;
+  private float qpsFraction;
```

Comment:
@bvaradar 
Part 1) I'm guessing you intended to compress them into a config called qpsForJobPerRegionServer. I feel fraction is more intuitive to understand and does not require clients to compute qpsForJobPerRegionServer by factoring qpsFraction and maxQpsPerRegionServer. Especially if clients end up using different values of maxQpsPerRegionServer in the math accidentally, it is not preferable. Note that this can also happen in the current method if the HoodieIndexConfig maxQpsPerRegionServer does not match across jobs, but that is easier to enforce since a) the documentation explicitly asks to make it same across all jobs. b) Clients can have a validation across their HoodieIndex configs. Also, note that with wrong math for qpsForJobPerRegionServer, clients could make the throttling useless (IOW set it to a very large value).

Part 2) We cannot have a aggregate QPS because that depends on how many Region Servers a Hbase table resides in, which is dynamic and under the control of HBase (a big table could span 100 regions and a small one can span 2). Hence numRegions a table resides in and hence the aggregate QPS are better fit for being factored within the library.

Reply:
@kaushikd49  Sounds fine.